xwiki-platform-core/pom.xml

```diff
@@ -125,9 +125,18 @@
 
                  Single justification example:
             -->
-            
-            
-            
+            <revapi.differences>
+              <justification>Needed changes for properly handling merge with fallback version.</justification>
+              <criticality>highlight</criticality>
+              <differences>
+                <item>
+                  <ignore>true</ignore>
+                  <code>java.method.addedToInterface</code>
+                  <new>method org.xwiki.store.merge.MergeManagerResult&lt;com.xpn.xwiki.objects.ElementInterface, java.lang.Object&gt; com.xpn.xwiki.objects.ElementInterface::merge(com.xpn.xwiki.objects.ElementInterface, com.xpn.xwiki.objects.ElementInterface, com.xpn.xwiki.doc.merge.MergeConfiguration, com.xpn.xwiki.XWikiContext)</new>
+                  <justification>This method replaces the now deprecated old ElementInterface#merge method.</justification>
```

Comment:
This is not the expected information, IMO. The point here is to explain why we are OK to break this API. So this should indicate why we don't think providing a default implementation is required (as it seems technically doable from what I see, it would just return a `MergeManagerResult` without Conflict, but it's less of a problem than a failing merge call).

Reply:
> it seems technically doable from what I see

yes it's technically doable, but I didn't want to do it as it wouldn't respect the contract of the config in case of the config is set to modify the current instance: calling it wouldn't have any effect, and the returned result would be wrong. So IMO it's better to just not implement it. 
I guess I can provide those info here.

Reply:
My main point is not that we should provide a default implementation (I don't think anyone implements this interface directly) but that what is indicated is not what is needed.

Reply:
Yes, we're never supposed to break backward compatibility. When we think we have no choice, this requires a VOTE, to make sure committers review it and agree (maybe some ideas will emerge to avoid the breakage). At the very least the VOTE needs to explain why it's ok to break it and why the impact is very limited.

Reply:
So I changed the revapi message, let me know if that's ok.

Reply:
Looks better yes. I guess I would have added a note indicating that the chance that this interface is directly implemented are close to 0 since it would not work in most cases.